runtime/compare.c

```diff
@@ -118,9 +136,11 @@ static intnat do_compare_val(struct compare_stack* stk,
 {
   struct compare_item * sp;
   tag_t t1, t2;
+  int signal_poll_timer = 0;
```

Comment:
counting down from `COMPARE_SIGNAL_POLL_PERIOD` is also an option. Then all the comparisons will be against 0. (this has no measurable impact, FWIW)

Reply:
On my machine counting down to 0 has a measurable impact, and the code is equally clear, so I went for it. (At the time of writing, the previous head with polling was 10% slower than the version without any polling logic, and the downto-0 version is only 5% slower.)